WireDomain/Sources/WireDomain/Synchronization/SyncManager.swift

```diff
@@ -32,20 +38,103 @@ protocol SyncManagerProtocol {
+    private let teamRepository: any TeamRepositoryProtocol
+    private let connectionsRepository: any ConnectionsRepositoryProtocol
+    private let conversationsRepository: any ConversationRepositoryProtocol
+    private let userRepository: any UserRepositoryProtocol
+    private let conversationLabelsRepository: any ConversationLabelsRepositoryProtocol
+    private let featureConfigsRepository: any FeatureConfigRepositoryProtocol
+    private let pushSupportedProtocolsUseCase: any PushSupportedProtocolsUseCaseProtocol
+    private let mlsProvider: MLSProvider
+    private let context: NSManagedObjectContext
+
+    // MARK: - Update event processor
+
     private let updateEventProcessor: any UpdateEventProcessorProtocol
 
+    // MARK: - Object lifecycle
+
     init(
         updateEventsRepository: any UpdateEventsRepositoryProtocol,
-        updateEventProcessor: any UpdateEventProcessorProtocol
+        teamRepository: any TeamRepositoryProtocol,
+        connectionsRepository: any ConnectionsRepositoryProtocol,
+        conversationsRepository: any ConversationRepositoryProtocol,
+        userRepository: any UserRepositoryProtocol,
+        conversationLabelsRepository: any ConversationLabelsRepositoryProtocol,
+        featureConfigsRepository: any FeatureConfigRepositoryProtocol,
+        updateEventProcessor: any UpdateEventProcessorProtocol,
+        pushSupportedProtocolsUseCase: any PushSupportedProtocolsUseCaseProtocol,
+        mlsProvider: MLSProvider,
+        context: NSManagedObjectContext
     ) {
         self.updateEventsRepository = updateEventsRepository
+        self.teamRepository = teamRepository
+        self.connectionsRepository = connectionsRepository
+        self.conversationsRepository = conversationsRepository
+        self.userRepository = userRepository
+        self.conversationLabelsRepository = conversationLabelsRepository
+        self.featureConfigsRepository = featureConfigsRepository
         self.updateEventProcessor = updateEventProcessor
+        self.pushSupportedProtocolsUseCase = pushSupportedProtocolsUseCase
+        self.mlsProvider = mlsProvider
+        self.context = context
+    }
+
+    func performSlowSync() async throws {
+        do {
+            try await updateEventsRepository.pullLastEventID()
+            try await teamRepository.pullSelfTeam()
+            try await teamRepository.pullSelfTeamRoles()
+            try await teamRepository.pullSelfTeamMembers()
+            try await connectionsRepository.pullConnections()
+            try await conversationsRepository.pullConversations()
+            try await userRepository.pullKnownUsers()
+            try await userRepository.pullSelfUser()
+            try await teamRepository.pullSelfLegalHoldStatus()
+            try await conversationLabelsRepository.pullConversationLabels()
+            try await featureConfigsRepository.pullFeatureConfigs()
+            try await pushSupportedProtocolsUseCase.invoke()
```

Comment:
I think this should not be a use case, but rather a method on `UserRepository`.

Reply:
There's already a method `pushSelfSupportedProtocols` in the `UserRepository`, this use case contains some business logic to calculate the supported protocols and then push them to server using the repo method. Repositories should not contain such business logic, so I would keep this object however maybe it doesn't to be public hence not be a use case and we could rename it `CalculateAndPushSupportedProtocols`.